e2e/scripts/smoke_test.sh

```diff
@@ -0,0 +1,112 @@
+#!/bin/bash -e
+
+test_source_repo=$(pwd)
+docker_test_repo=/workspace/go/src/github.com/rook/rook
+git_smoke_test_directory=github.com/rook/rook/e2e/tests/smoke
+container_image=quay.io/quantum/rook-test
+tmp_docker_sock_path=/tmp/docker.sock
+results_dir=results
+results_filename=test.log
+
+#create the rook infrastructure container
+rook_infra::create() {
+    export id=$(docker run \
+        --net=host \
+        -d \
+        -e GOPATH=/workspace/go \
+        -e GOROOT=/usr/lib/go \
+        --privileged \
+        --security-opt=seccomp:unconfined \
+        -v /var/run/docker.sock:${tmp_docker_sock_path} \
+        -v ${test_source_repo}:${docker_test_repo} \
+        -w ${docker_test_repo}/e2e \
+        ${container_image} \
+        /sbin/init)
+
+        rc=$?; if [[ $rc != 0 ]]; then set -e; fi
+}
+
+#prepare the rook_infra container to run tests
+rook_infra::init() {
+    echo Creating results directory
+    docker exec ${id} mkdir -p ${results_dir}
+    rc=$?; if [[ $rc != 0 ]]; then set -e; fi
+
+    echo Removing infra docker.sock
+    docker exec ${id} rm -rfv /var/run/docker.sock
+    rc=$?; if [[ $rc != 0 ]]; then set -e; fi
+
+    echo Creating sysmlink to host docker.sock
+    docker exec ${id} ln -s ${tmp_docker_sock_path} /var/run/docker.sock
+    rc=$?; if [[ $rc != 0 ]]; then set -e; fi
+    echo Success...
+
+    echo Installing rook-test-framework dependencies...
+    docker exec ${id} /bin/bash -c \
+    "glide install"
+    rc=$?; if [[ $rc != 0 ]]; then set -e; fi
+    echo Success...
+}
+
+rook_infra::run_test() {
+    local test_name_regex=$1
+    local tag_name=$2
+    local rook_platform=$3
+    local k8s_version=$4
+
+    docker exec -t ${id} /bin/bash -c \
+        "go test -timeout 1200s -run ${test_name_regex} ${git_smoke_test_directory} --rook_platform=${rook_platform} --k8s_version=${k8s_version} --rook_version=${tag_name} -v 2>&1 | tee ${results_dir}/${results_filename}"
+
+    rc=$?; if [[ $rc != 0 ]]; then set -e; fi
+}
+
+rook_infra::gather_results() {
+    echo Gathering results...
+
+    #install junit parser && create junit results
+    docker exec ${id} /bin/bash -c \
+        "go get -u -f github.com/jstemmer/go-junit-report && cat ${results_dir}/test.log | go-junit-report > ${results_dir}/junit.xml"
+
+    rc=$?; if [[ $rc != 0 ]]; then set -e; fi
+}
+
+rook_infra::cleanup() {
+    echo Removing rook-test-framework container...
+    docker kill ${id}
+    docker rm ${id}
+}
+
+{ #try
+
+    if [ -z "$1" ]; then
```

Comment:
this parameter parsing implementation is a bit fragile, where it won't work if the args are passed in an order different than what the script expects or if the caller fails to pass all 3 args.  It would be nice if the caller could only pass the single arg they want to override and then defaults would be used for the rest.  That's not possible with this implementation.

A better implementation would be to cycle through the args with `shift` use defaults for any args not specified (e.g., https://unix.stackexchange.com/a/122848).

Consider opening an issue to improve this later.